### PR TITLE
Begin exposing prometheus metrics.

### PR DESCRIPTION
**Pull request by dgoodwin**
- Run prometheus metrics endpoint on port 2112. Vendor prometheus libraries.
- Metrics for all jobs, periods, and releases.

Metrics appear as:

```
sippy_job_pass_ratio{name="release-openshift-ocp-installer-e2e-metal-serial-4.10",period="current",release="4.10"} 0.045454545454545456
sippy_job_pass_ratio{name="release-openshift-ocp-installer-e2e-metal-serial-4.10",period="previous",release="4.10"} 0.36363636363636365
sippy_job_pass_ratio{name="release-openshift-ocp-installer-e2e-metal-serial-4.10",period="twoDay",release="4.10"} 0
```

For each job. Future enhancements will be additional labels for blocking/informing (we don't really know this right now), frequent/infrequent, and an owner for routing purposes.

### CODE REVIEW

**Comment by stbenjam** [comment]
Neat! Code looks straightforward enough, just needs go mod fixing

File: resources/deploymentconfig-historical.yaml [hunk 1]
```diff
@@ -51,6 +51,9 @@ spec:
         ports:
         - containerPort: 8080
           protocol: TCP
+        - name: metrics
+          containerPort: 2112
+          protocol: TCP
```
Comment (stbenjam) [inline resources/deploymentconfig-historical.yaml:56]: I don't think we care about metrics on the historical instance (https://sippy-historical-bparees.apps.ci.l2s4.p1.openshiftapps.com/).  WDYT?

Comment (dgoodwin) [reply resources/deploymentconfig-historical.yaml:56]: Probably not, I'll remove.

**Pull request merged**